lib/pbench/agent/tool_data_sink.py

```diff
@@ -1243,12 +1243,12 @@ def _valid_data(self, data):
                 return (action, directory, args)
 
     @staticmethod
-    def check_tms(tms):
-
+    def check_failed_tool_init(tms: dict) -> bool:
+        """checks if there are any tool failures in the Tool-Meisters"""
         for host, tm in tms.items():
             if len(tm["failed_tools"]) > 0:
-                return "failure"
-        return "success"
+                return False
+        return True
```

Comment:
Since you never reference `host`, you might as well omit it:
```
        for _, tm in tms.items():
```
or, better still
```
        for tm in tms.values():
```